src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -314,5 +315,91 @@ private async Task AssertApprover(ClaimsPrincipal user, ReviewRevisionModel revi
                 throw new AuthorizationFailedException();
             }
         }
+
+        private async Task<ReviewModel> GetMasterReviewAsync(ClaimsPrincipal user, string language, string packageName)
+        {
+            if (user == null)
+            {
+                throw new UnauthorizedAccessException();
+            }
+
+            var review = await _reviewsRepository.GetMasterReviewForPackageAsync(language, packageName);
+            if (review != null)
+            {
+                review.UpdateAvailable = review.Revisions
+                .SelectMany(r => r.Files)
+                .Any(f => f.HasOriginal && GetLanguageService(f.Language).CanUpdate(f.VersionString));
+            }
+
+            return review;
+        }
+
+        public async Task<ReviewModel> CreateMasterReviewAsync(ClaimsPrincipal user, string originalName, string label, Stream fileStream, bool runAnalysis)
+        {
+            var revision = new ReviewRevisionModel();
+
+            //Create code file for uploaded package
+            ReviewCodeFileModel codeFile = await CreateFileAsync(
+                revision.RevisionId,
+                originalName,
+                fileStream,
+                runAnalysis);
+
+            revision.Files.Add(codeFile);
+            revision.Author = user.GetGitHubLogin();
+            revision.Label = label;
+
+            //Get current review for package
+            var review = await GetMasterReviewAsync(user, codeFile.Language, codeFile.PackageName);
+            if (review != null)
+            {
+                // Check if urrent review needs to be updated to rebuild using latest language processor
+                if (review.UpdateAvailable)
+                {
+                    await UpdateReviewAsync(user, review.ReviewId);
+                }
+
+                //Get latest revision of reviw and check diff
+                var lastRevision = review.Revisions.Last();
+                var lastRevisionFile = await _codeFileRepository.GetCodeFileAsync(lastRevision);
+                var lastHtmlLines = lastRevisionFile.RenderReadOnly(false);
+                var lastRevisionTextLines = lastRevisionFile.RenderText(false);
+
+                var renderedCodeFile = await _codeFileRepository.GetCodeFileAsync(revision);
+                var fileTextLines = renderedCodeFile.RenderText(false);
+                var fileHtmlLines = renderedCodeFile.RenderReadOnly(false);
+
+                var diffLines = InlineDiff.Compute(
```

Comment:
Can you just directly compare text render output?

Reply:
Reason for not using current diff is to get formatted added and removed view If we eventually need to show only diff in CI logs. I will check and make this change.